sunpy/net/dataretriever/sources/goes.py

```diff
@@ -58,18 +58,28 @@ def _get_url_for_timerange(self, timerange, **kwargs):
             Data type to return for the particular GOES satellite. Supported
             types depend on the satellite number specified. (default = xrs_2s)
         """
-
+        start_date = parse_time(timerange.start)
+        end_date = parse_time(timerange.end)
```

Comment:
why do we need to `parse_time` somehting it's already  at `TimeRange`? I imagine it's not needed, but maybe I'm missing some edge case - in which probably should be added to the TimeRange implementation.